### PR TITLE
handle types in docstring args, Exception types

### PR DESCRIPTION
**Pull request by cs01**
Fix 1:
The signature cannot be determined from this code.
```
class MyError(ValueError):
    """Raise when trying to ..."""

    pass
```

 This PR uses an empty signature when signature cannot be determined. 

Fix 2:
When a docstring for arguments has a type hint, an error would be thrown. This PR strips out text in parentheses from the argument name, and allows documentation to be successfully generated.

(Unfortunately I do not have the time to write full tests for these fixes, but wanted to share them anyway)

### CODE REVIEW

File: src/mkdocstrings/documenter.py [hunk 1]
```diff
@@ -342,12 +342,13 @@ def get_class_documentation(self, class_: Type[Any], module: Optional[ModuleType
...
-            path=path,
-            file_path=file_path,
-            docstring=Docstring(textwrap.dedent(class_.__doc__ or ""), inspect.signature(class_)),
-        )
+        try:
+            signature = inspect.signature(class_)
+        except ValueError:
+            print(f"Failed to get signature for {class_name}")
+            signature = inspect.Signature()
+        docstring = Docstring(textwrap.dedent(class_.__doc__ or ""),)
```
Comment (pawamoy) [inline src/mkdocstrings/documenter.py]: I think you forgot to pass the `signature` to the `Docstring` constructor 😛 ?

File: src/mkdocstrings/docstrings.py [hunk 2]
```diff
@@ -211,7 +211,18 @@ def read_parameters_section(self, lines, start_index):
         parameters = []
         block, i = self.read_block_items(lines, start_index)
         for param_line in block:
-            name, description = param_line.lstrip(" ").split(":", 1)
+            try:
+                name_with_type, description = param_line.lstrip(" ").split(":", 1)
+                paren_index = name_with_type.find("(")
+                if paren_index != -1:
+                    # name (type)
+                    name = name_with_type[0:paren_index].strip()
```
Comment (pawamoy) [inline src/mkdocstrings/docstrings.py]: I like that, it could help implementing #27

Comment (cs01) [reply src/mkdocstrings/docstrings.py]: Definitely, would be easy to add a regex to extract the type if it were present.

File: src/mkdocstrings/documenter.py (hunk 1 above)
Comment (cs01) [reply src/mkdocstrings/documenter.py]: Thanks, fixed

**Comment by pawamoy** [comment]
Alright, there's a conflict to resolve and we're good to go!

**Comment by pawamoy** [comment]
Thank you @cs01!

**Pull request merged**